url8d1df200a8_gilleain_signatures_tgz-pJ8-signature_chemistry_LargeMoleculeTestJ8/src/signature/chemistry/MoleculeFactory.java

```diff
@@ -0,0 +1,140 @@
+        molecule.addSingleBond(0, 2);
+        molecule.addSingleBond(1, 2);
+        return molecule;
+    }
+    
+    public static Molecule fourCycle() {
+        Molecule molecule = new Molecule("C", 4);
+        molecule.addSingleBond(0, 1);
+        molecule.addSingleBond(0, 3);
+        molecule.addSingleBond(1, 2);
+        molecule.addSingleBond(2, 3);
+        return molecule;
+    }
+    
+    public static Molecule fiveCycle() {
+        Molecule molecule = new Molecule("C", 5);
+        molecule.addSingleBond(0, 1);
+        molecule.addSingleBond(0, 4);
+        molecule.addSingleBond(1, 2);
+        molecule.addSingleBond(2, 3);
+        molecule.addSingleBond(3, 4);
+        return molecule;
+    }
+    
+    public static Molecule sixCycle() {
+        Molecule molecule = new Molecule("C", 6);
+        molecule.addSingleBond(0, 1);
+        molecule.addSingleBond(0, 5);
+        molecule.addSingleBond(1, 2);
+        molecule.addSingleBond(2, 3);
+        molecule.addSingleBond(3, 4);
+        molecule.addSingleBond(4, 5);
+        return molecule;
+    }
+    
+    public static Molecule threeStar() {
+        Molecule molecule = new Molecule("C", 4);
+        molecule.addSingleBond(0, 1);
+        molecule.addSingleBond(0, 2);
+        molecule.addSingleBond(0, 3);
+        return molecule;
+    }
+    
+    public static Molecule fourStar() {
+        Molecule molecule = new Molecule("C", 5);
+        molecule.addSingleBond(0, 1);
+        molecule.addSingleBond(0, 2);
+        molecule.addSingleBond(0, 3);
+        molecule.addSingleBond(0, 4);
+        return molecule;
+    }
+    
+    public static Molecule fiveStar() {
+        Molecule molecule = new Molecule("C", 6);
+        molecule.addSingleBond(0, 1);
+        molecule.addSingleBond(0, 2);
+        molecule.addSingleBond(0, 3);
+        molecule.addSingleBond(0, 4);
+        molecule.addSingleBond(0, 5);
+        return molecule;
+    }
+    
+    public static Molecule propellane() {
+        Molecule molecule = new Molecule("C", 5);
+        molecule.addSingleBond(0, 1);
+        molecule.addSingleBond(0, 4);
+        molecule.addSingleBond(1, 2);
+        molecule.addSingleBond(1, 3);
+        molecule.addSingleBond(1, 4);
+        molecule.addSingleBond(2, 4);
+        molecule.addSingleBond(3, 4);
+        return molecule;
+    }
+    
+    public static Molecule methylatedCyclobutane() {
+        Molecule molecule = new Molecule("C", 5);
+//        molecule.addSingleBond(0, 1);
+//        molecule.addSingleBond(1, 2);
+//        molecule.addSingleBond(1, 3);
+//        molecule.addSingleBond(1, 4);
+//        molecule.addSingleBond(2, 4);
+//        molecule.addSingleBond(3, 4);
+        molecule.addSingleBond(0, 1);
+        molecule.addSingleBond(0, 2);
+        molecule.addSingleBond(0, 3);
+        molecule.addSingleBond(0, 4);
+        molecule.addSingleBond(2, 3);
+        molecule.addSingleBond(3, 4);
+        return molecule;
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 97:110, 114:121.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.